go-controller/pkg/ovn/external_ids_syncer/port_group/port_group_sync.go

```diff
@@ -0,0 +1,360 @@
+	ingressDefaultDenySuffix = "ingressDefaultDeny"
+	// egressDefaultDenySuffix is the suffix used when creating the ingress port group for a namespace
+	egressDefaultDenySuffix      = "egressDefaultDeny"
+	defaultNetworkControllerName = "default-network-controller"
+	// values for libovsdbops.PolicyDirectionKey
+	// We don't reuse any external constants in this package, as it should update db entries from a
+	// pre-defined format. If in the future values for some external constants change, this package shouldn't be affected.
+	policyDirectionIngress = "Ingress"
+	policyDirectionEgress  = "Egress"
+)
+
+type updatePortGroupInfo struct {
+	acls  []*nbdb.ACL
+	oldPG *nbdb.PortGroup
+	newPG *nbdb.PortGroup
+}
+
+type PortGroupSyncer struct {
+	nbClient    libovsdbclient.Client
+	getPGWeight func(acls, ports int) float64
+}
+
+// getPGWeight returns weight of a port group based on the number of linked acls and ports.
+// port group syncer updates both port groups and related acls, transaction time depends on the number of acls related to every port group.
+// The concept of "weight" reflects that port group with 1 ACL and port group with 100 ACLs will have different
+// transaction time. Based on local testing, the results are
+// Update for 30`000 port groups with 1 ACL each takes 7,14 sec
+// here almost linear dependency begins
+// Update for 5`000 port groups with 10 ACL each takes 6 sec
+// Update for 500 port groups with 100 ACL each takes 5,95 sec
+// Update for 50 port groups with 1000 ACL each takes 5,84 sec
+// Update for 5 port groups with 10000 ACL each takes 5,42 sec
+// Considering given times safe within 10 second timeout, 5 port groups with 10000 ACLs may be updated in one batch.
+// That makes a weight of port group with 10000 ACLs = 1/5.
+// The number of ports in a port group also affects the weight. By adding a given number of ports in the local testing
+// we got the following extra transaction time (o ports = 0 extra time):
+// 50 pg + 1000 ACL:
+//
+//	 1K ports => 0.69 s
+//	 5K ports => 2.89 s
+//	10K ports => 6.85 s
+//
+// 500 pg + 100 ACL:
+//
+//	 1K ports => 5.27 s
+//	 5K ports => 30.39 s
+//	10K ports => 60.5 s
+//
+// 5000 pg + 10 ACL:
+//
+//	100 ports => 5.9 s
+//	200 ports => 16.2 s
+//	 1K ports => 58.5 s
+//
+// Considering 7 seconds is a safe transaction time, the following approximation may be applied:
+// 1000 ports * 500 pg <= 7 seconds extra (weight = 1)
+// => 1 port extra weight = 1/(500 * 1000) = 0.000002
+func getPGWeight(acls, ports int) float64 {
+	portsWeight := 0.000002 * float64(ports)
+
+	if acls < 10 {
+		// extra coefficient for a small number of acls
+		if acls == 0 {
+			// updating a port group without acls may be estimated with a port group with 1 acl update time
+			acls = 1
+		}
+		return portsWeight + float64(acls)/30000
+	}
+	// for acls > 10, use linear dependency
+	return portsWeight + float64(acls)/50000
```

Comment:
Why don't you multiply by `0.000003` and `0.000002` as you did with the pot weight? Or divide `/50000` for port weight? Just wondering....

Reply:
because I wrote it on different days :D 
I guess 1/3 doesn't look good if I write it in float64, so maybe default to division?